app/Fission/Internal/Orphanage/RIO.hs

```diff
@@ -20,3 +25,35 @@ instance Has (DB.Pool PG) cfg => MonadSelda (RIO cfg) where
   withConnection action = do
     DB.Pool pool <- Config.get
     withResource pool action
+
+instance 
+  ( HasProcessContext cfg
+  , HasLogFunc cfg
+  , Has IPFS.BinPath cfg
+  , Has IPFS.Timeout cfg
+  )
+  => MonadLocalIPFS (RIO cfg) where
+    withIPFSProc processor inStream outStream opts = do
+      IPFS.BinPath ipfs <- Config.get
+      IPFS.Timeout secs <- Config.get
+      let opts' = ("--timeout=" <> show secs <> "s") : opts
+      proc ipfs opts' <| processor
+                      . setStdin  inStream
+                      . setStdout outStream
+
+    ipfsRun opts arg = withIPFSProc readProcess (byteStringInput arg) byteStringOutput opts
+
+    getTimeout = Config.get
+
+instance 
+  ( Has IPFS.URL cfg
+  , Has HTTP.Manager cfg
+  )
+  => MonadRemoteIPFS (RIO cfg) where
+    runRemote query = do
+      IPFS.URL url            <- Config.get
+      manager :: HTTP.Manager <- Config.get
+      url
+        |> mkClientEnv manager
+        |> runClientM query
+        |> liftIO
```

Comment:
Yeah, this file belongs in the app. People use `RIO` in many different ways. Totally get that you need this to test development, though

Reply:
You could also create a `newtype` for testing